src/daft-core/src/array/ops/utf8.rs

```diff
@@ -1339,41 +1339,49 @@ impl Utf8Array {
     }
 
     pub fn normalize(&self, opts: Utf8NormalizeOptions) -> DaftResult<Utf8Array> {
-        let whitespace_regex = regex::Regex::new(r"\s+").unwrap();
-
-        let arrow_result = self
-            .as_arrow()
-            .iter()
-            .map(|maybe_s| {
+        Ok(Utf8Array::from_iter(
+            self.name(),
+            self.as_arrow().iter().map(|maybe_s| {
                 if let Some(s) = maybe_s {
-                    let mut s = s.to_string();
+                    let mut s = if opts.white_space {
+                        s.trim().to_string()
+                    } else {
+                        s.to_string()
+                    };
 
-                    if opts.remove_punct {
-                        s = s.chars().filter(|c| !c.is_ascii_punctuation()).collect();
-                    }
+                    let mut prev_white = true;
+                    s = s
+                        .chars()
+                        .filter_map(|c| {
```

Comment:
Wonder if this would be even faster if we have the common case first (no whitespace and isn't ascii punctuation), followed by whitespace and then lastly by punctuation.

Reply:
Good call, got around a 7% speedup from that